File: code/app/src/main/java/in/curioustools/menu_maker/db/MenuDB.java
package in.curioustools.menu_maker.db;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import in.curioustools.menu_maker.modal.MenuEntry;


@Database(entities = MenuEntry.class,version = 1,exportSchema = false)
public  abstract class MenuDB extends RoomDatabase{
    abstract MenuTableAccessDao getSongsTableAccessDao();

    private static volatile MenuDB INSTANCE = null;
    private static final  String DB_NAME = "MENU.db";

    @NonNull
    static synchronized MenuDB getInstance(final Context context) {

        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context, MenuDB.class, DB_NAME).build();
        }
        return INSTANCE;

        //For testing:INSTANCE = Room.inMemoryDatabaseBuilder(context, MenuDB.class).build();

    }


}

File: code/app/src/main/java/in/curioustools/menu_maker/db/MenuDB.kt
package `in`.curioustools.menu_maker.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [MenuEntry::class], version = 2, exportSchema = false)
abstract class MenuDB : RoomDatabase() {
    abstract val songsTableAccessDao: MenuTableAccessDao?

    companion object {
        @Volatile
        private var INSTANCE: MenuDB? = null
        private const val DB_NAME = "MENU.db"

        @JvmStatic @Synchronized
        fun getInstance(context: Context?): MenuDB {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(
                    context!!,
                    MenuDB::class.java,
                    DB_NAME
                ).fallbackToDestructiveMigration().build()
            }
            return INSTANCE!!

            //For testing:INSTANCE = Room.inMemoryDatabaseBuilder(context, MenuDB.class).build();
        }
    }
}